### PR TITLE
Settings branch

### PR DESCRIPTION
**Pull request by korneliarohulko**
Settings page configuration

### CODE REVIEW

File: client/js/Utils/SettingsUtils.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+export default class SettingsUtils extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      language: T.locale
+    };
+  }
+
+  componentDidMount() {
+    this.checkEvent();
```
Comment (osdnk) [inline client/js/Utils/SettingsUtils.js]: It could me one `async componentDidMount` fucntion IMO

File: client/js/Utils/SettingsUtils.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    const eventNotification = await ClientStorage.getEventNotification();
+    if (eventNotification)
+      this.setState({ eventNotification: eventNotification });
+    else
+      this.setState({ eventNotification: false });
+  }
+
+  async checkNotificationAd() {
+    const adNotification = await ClientStorage.getAdNotification();
+    if (adNotification)
```
Comment (osdnk) [inline client/js/Utils/SettingsUtils.js]: `this.setState({adNotification: adNotification})`
Isn't it the same as this whole `if` statement?

File: client/js/Utils/ClientStorage.js [hunk 3]
```diff
@@ -8,4 +8,28 @@ export default class ClientStorage {
...
+
+  static async getEventNotification() {
+    return JSON.parse(await AsyncStorage.getItem('eventNotification'));
+  }
+
+  static setAdNotification(adNotification) {
+    AsyncStorage.setItem('adNotification', JSON.stringify(adNotification));
+  }
+
+  static async getAdNotification() {
```
Comment (osdnk) [inline client/js/Utils/ClientStorage.js]: Ad?

File: client/js/Translation/Translations/pl.js [hunk 4]
```diff
@@ -30,5 +27,10 @@ export default {
     appBody: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
     contactHeader: 'Kontakt',
     contactBody: 'Lorem ipsum dolor sit amet'
+  },
+  settings: {
+    adLabel: 'Notyfikacje ogłoszeń',
```
Comment (osdnk) [inline client/js/Translation/Translations/pl.js]: Przypomnienia sounds better for me

File: client/js/Screens/Settings/SettingsScreen.js [hunk 5]
```diff
@@ -1,55 +1,10 @@
...
-              {T.getTranslation('bottomNavBar.news')}
-            </Text>
-          </TranslatableComponent>
-        </ScrollView>
-      </Scene>
-    );
-  }
-}
+const SettingsScreen = ({ navigation }) => (
+  <SettingsUtils/>
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: hmhmhm. Isn't it pointless to make component only to be wrapper for another? 🤔

Comment (korneliarohulko) [reply client/js/Screens/Settings/SettingsScreen.js]: I have followed QRScreen concept to split into Screen&Utils just to keep Screen clean, but I wasn't convinced frankly speaking. How it should be done then?

File: client/js/Utils/ClientStorage.js (hunk 3 above)
Comment (korneliarohulko) [reply client/js/Utils/ClientStorage.js]: reminderNotification? I have no idea for this getter/setter name

**Comment by osdnk** [comment]
Why dont u just paste code from this component here? And change this SFC
into normal class?



2017-12-01 18:09 GMT+01:00 korneliarohulko <notifications@github.com>:

**Comment by osdnk** [comment]
sounds ok



2017-12-01 18:11 GMT+01:00 korneliarohulko <notifications@github.com>:

File: client/js/Screens/Settings/SettingsScreen.js (hunk 5 above)
Comment (osdnk) [reply client/js/Screens/Settings/SettingsScreen.js]: Why dont u just paste code from this component here? And change this SFC
into normal class?

File: client/js/Screens/Settings/SettingsScreen.js [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
 import { IntlText } from '../../Translation/IntlText';
-import TranslatableComponent from '../../Translation/TranslatableComponent';
+import T from '../../Translation/Translator';
+import Switch from '../../GlamorousComponents/Switch';
+import ClientStorage from '../../Utils/ClientStorage';
+import SettingsCard from '../../GlamorousComponents/SettingsCard';
 import DesignUtil from '../../Utils/DesignUtils';
 
+const styles = StyleSheet.create({
+  text: {
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: nah, to Glamorous/config

File: client/js/Screens/Settings/SettingsScreen.js [hunk 7]
```diff
@@ -1,12 +1,23 @@
...
+    flex: 0.8
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: strange. Use rather `space-beetwen` on my screen component is cut a bit :)

File: client/js/Screens/Settings/SettingsScreen.js [hunk 8]
```diff
@@ -1,12 +1,23 @@
...
+  },
+  picker: {
+    flex: 0.5
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: same

File: client/js/Screens/Settings/SettingsScreen.js [hunk 9]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
   }
 
-  switchLanguage() {
-    let index = SupportedLanguages.indexOf(this.state.language);
-    index = (index + 1) % SupportedLanguages.length;
-    const newLanguage = SupportedLanguages[index];
-    this.setState({ language: newLanguage });
-    T.switchLanguage(newLanguage);
+  async componentDidMount() {
+    const eventNotification = await ClientStorage.getEventNotification();
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: great!

File: client/js/Screens/Settings/SettingsScreen.js [hunk 10]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+      this.setState({ timeRemindValue: 5 });
+    }
+
+    if (newsNotification)
+      this.setState({ newsNotification: newsNotification });
+    else
+      this.setState({ newsNotification: false });
   }
 
+  onValueChange = (value) => {
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: `Value` is not a good name. It tells nothing

File: client/js/Screens/Settings/SettingsScreen.js [hunk 11]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+  };
+
+  newsNotificationChange = () => {
+    ClientStorage.setNewsNotification(!this.state.newsNotification);
+    this.setState({
+      newsNotification: !this.state.newsNotification
+    });
+  };
+
+  timeRemindValueChange = (value) => {
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: let us imagine that here's Java. Is this method `private`?  If yes, then name it with `_` : `_newsNotificationChange`

File: client/js/Screens/Settings/SettingsScreen.js [hunk 12]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+    ClientStorage.setTimeRemindValue(value);
+    this.setState({
+      timeRemindValue: value
+    });
+  };
+
   render() {
+    const mapTiming = {
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: You have a dict. Then you take every key of this dict and map it to value 🤔 why? `times=[5, 10, 30]; times.map(t=>...)`

File: client/js/Screens/Settings/SettingsScreen.js [hunk 13]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+      '5 minutes before': 5,
+      '10 minutes before': 10,
+      '30 minutes before': 30
+    };
+
     return (
-      <Scene style={{ alignItems: 'center' }}>
+      <Scene style={{flex: 1}}>
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: `<Scene>`

File: client/js/Screens/Settings/SettingsScreen.js [hunk 14]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
-              {T.getTranslation('bottomNavBar.news')}
-            </Text>
-          </TranslatableComponent>
+          <SettingsCard>
+            <IntlText id='settings.adLabel' style={styles.text}></IntlText>
+            <Switch checked={this.state.newsNotification}
+              onCheckedChange={() => this.newsNotificationChange()}/>
+          </SettingsCard>
+          <SettingsCard>
+            <IntlText id='settings.remindLabel' style={styles.text}></IntlText>
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: dont use `local` stylesheets

File: client/js/Screens/Settings/SettingsScreen.js [hunk 15]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+            <Switch checked={this.state.eventNotification}
+              onCheckedChange={() => this.eventNotificationChange()}
+            />
+          </SettingsCard>
+          {this.state.eventNotification ? <Card><Picker style={styles.picker}
+            selectedValue = {this.state.timeRemindValue}
+            onValueChange = {(value) => {this.timeRemindValueChange(value)}}>
+            { Object.keys(mapTiming).map( value => {
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: -

File: client/js/Screens/Settings/SettingsScreen.js [hunk 16]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+              return (<Picker.Item key={value} label={value} value={mapTiming[value]} />); } )}
+          </Picker></Card> : null}
+          <SettingsCard>
+            <IntlText id='settings.language' style={styles.picker}></IntlText>
+            <Picker style={styles.picker}
+              selectedValue = {this.state.language}
+              onValueChange = {(value) => {this.onValueChange(value)}}>
+              <Picker.Item label='en' value='en' />
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: I think you should map values from translator/config. It seems to be hardcoded. What if we decide to provide esperanto as well? :)

File: client/js/Screens/Settings/SettingsScreen.js [hunk 17]
```diff
@@ -15,36 +26,91 @@ class SettingsScreen extends React.Component {
...
+            />
+          </SettingsCard>
+          {this.state.eventNotification ? <Card><Picker style={styles.picker}
+            selectedValue = {this.state.timeRemindValue}
+            onValueChange = {(value) => {this.timeRemindValueChange(value)}}>
+            { Object.keys(mapTiming).map( value => {
+              return (<Picker.Item key={value} label={value} value={mapTiming[value]} />); } )}
+          </Picker></Card> : null}
+          <SettingsCard>
+            <IntlText id='settings.language' style={styles.picker}></IntlText>
```
Comment (osdnk) [inline client/js/Screens/Settings/SettingsScreen.js]: Too many in one line. Space is cheap :)

**Pull request merged**